### PR TITLE
Migrate Onfido go library to v3

### PR DESCRIPTION
**Pull request by HizzleDev**
This PR adds in all the necessary changes and fixes the tests for the migration to v3. Our interface is not completely backwards compatible so we may have to make use of the semver versioning in Go modules (I'm not sure if we're doing this with tags currently).

Migration guide is found here: https://developers.onfido.com/guide/api-v2-to-v3-migration-guide

### CODE REVIEW

File: webhook_test.go [hunk 1]
```diff
@@ -45,7 +45,7 @@ func TestValidateSignature_InvalidSignature(t *testing.T) {
 
 func TestValidateSignature_ValidSignature(t *testing.T) {
 	wh := webhook{Token: "abc123"}
-	err := wh.ValidateSignature([]byte("hello world"), "fcc98c5b4f306cfe6b5b8fcce03ddb33fc13ae6b")
+	err := wh.ValidateSignature([]byte("hello world"), "8c301acf7e955038b486de8f2a35f7f28bb5755fd1f77e1dbf9ef9e27713ad0d")
```
Comment (HizzleDev) [inline webhook_test.go:48]: Keen to know if these tests were working before @crlssn 

It seemed to me that the signature that it was matching was wrong and I'm assuming they were failing before?

**Review by alexykot — APPROVED** [review]
Mechanically this LGTM. 
I've not checked this against the migration guide, so have no opinion on whether this meets requirements for v3 integration.

**Review by jacobcartledge — APPROVED** [review]
Checked as much as i can against the migration guide. All LGTM.

**Review by crlssn — CHANGES_REQUESTED** [review]
I think you need to update `DefaultEndpoint` to `https://api.onfido.com/v3.1`.

File: webhook_test.go (hunk 1 above)
Comment (crlssn) [reply webhook_test.go:48]: Yeah, they are failing in `master` as well.

**Review by crlssn — APPROVED** [review]
Never mind, you had :D

**Comment by crlssn** [comment]
Merge and tag it as `v1.0.0`?

**Pull request merged**